internal/controllers/clustercatalog_controller.go

```diff
@@ -70,9 +77,8 @@ func (r *ClusterCatalogReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	catalogFsys, err := r.CatalogCache.Get(existingCatalog.Name, existingCatalog.Status.ResolvedSource.Image.Ref)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error retrieving cache for catalog %q: %v", existingCatalog.Name, err)
-	}
-	if catalogFsys != nil {
+		l.Info("retrying cache population: found previous error from catalog cache", "cacheErr", err)
```

Comment:
@joelanford Just noticed this PR. We did this intentionally so we propagate the cache fetch error into `ClusterExtension`.

As far as I understand with this change we will try to populate the cache from both cluster catalog controller and from this controller. Is this what we want to do?


Refs:

* https://github.com/operator-framework/operator-controller/pull/1318
* https://github.com/operator-framework/operator-controller/pull/1284#discussion_r1767560927